### PR TITLE
Document dg namespace

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Some functions don't show up in the Doxygen search if the namespace isn't documented.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsdeppe — APPROVED** [review]

**Review by moxcodes — DISMISSED** [review]

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp [hunk 1]
```diff
@@ -19,6 +19,12 @@
 #include "NumericalAlgorithms/Spectral/Projection.hpp"
 #include "Options/Options.hpp"
 
+/// Functionality related to discontinuous Galerkin schemes
+namespace dg {
+// The classes below should be moved in the `dg` namespace. See issue:
+// https://github.com/sxs-collaboration/spectre/issues/2140
```
Comment (nilsdeppe) [inline src/NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp]: @wthrowe requested this comment be removed in #1725, so please remove it here too

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp (hunk 1 above)
Comment (nilsvu) [reply src/NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp]: In that PR the comment in question referred to the PR itself. Linking to issues is a practice we have been encouraging I think. Thoughts?

Comment (nilsvu) [reply src/NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp]: Oh you referred to a comment I had not seen until now. I remove it then. Does that mean you think we shouldn’t link to issues from code @wthrowe @nilsdeppe @kidder? I see no harm in making the relevant issue easier to find from the code

**Review by moxcodes — DISMISSED** [review]

**Comment by nilsvu** [comment]
@wthrowe could you take a look at this PR, since it's a commit from #1725 that you commented on?

**Comment by wthrowe** [comment]
Isn't this waiting for others to weigh in on my comment?

I don't like the change you made, though.  I don't object to the link, except in that I don't think there should be a comment at all.  If we do keep the comment I think the link should stay in.

**Comment by nilsdeppe** [comment]
I agree with @wthrowe, if we have an issue we don't need a comment in the code. There might be some cases where it's useful, e.g. when writing a detailed description of what's going on to say `Issue #BLAH is related to this` if the issue doesn't address the thing being commented on. Having the comment in the code is basically having a `TODO` without a `TODO`, and (I think) we decided long ago that issues are preferred over `TODO`s in the code because those `TODO`s will (generally) never get done because they get lost.

**Comment by nilsvu** [comment]
Ok, removed the comment altogether then. Looks good?

**Review by nilsdeppe — APPROVED** [review]
I'm at least happy with this :)

**Review by moxcodes — APPROVED** [review]

**Comment by nilsvu** [comment]
@wthrowe please confirm you approve of this change :)

**Review by wthrowe — APPROVED** [review]

**Pull request merged**